### ISSUE TITLE
Shopping cart bug tracker

### ISSUE THREAD

**Issue by Gibbsdavidl** [ImagingDataCommons/IDC-WebApp#1401]
General issue to track bugs and issues related to the shopping cart.

![image](https://github.com/user-attachments/assets/62fcbc50-9fda-42c3-9aed-ddfd161bb65e)

**Comment by Gibbsdavidl** [comment]
Cart bug #1

1. Under FDA, select VICTRE, open chevron in Collections to open Cases
2. Click on case level cart … nothing happens
3. Click again… now it's selected.

Expectation is that it would be selected the first time it's clicked.

**Comment by Gibbsdavidl** [comment]
Cart observation 1

After opening cases and making independent selections across pages, I was surprised to find that when a new case chevron was opened, the past chevrons were not closed. This leads to a collection of studies. 

My intuition was that each time a case chevron was opened, what was displayed in the studies was only the last case that was opened.

Here's an image after making a number of selections across pages:
![image](https://github.com/user-attachments/assets/8db2ab74-474a-4570-986f-6fff04de7a6b)

**Comment by Gibbsdavidl** [comment]
Cart observation 2  (or bug?)

1. make some cart selections
2. open the cart screen
3. then click on 'explore images' button in the top menu
4. the cart is now empty

Expectation is that there would be no changes to the cart.

**Comment by fedorov** [comment]
@G-White-ISB the explore page looks really nice! I would actually love for that UI facelift to just go to the main portal without waiting for the cart to be finished (but definitely let's not make anyone's life even more complex by separating those, unless its not difficult).

I spent some time with this today, and here are some findings. Some of those are unfortunately seem to be rather serious.

* For the tooltip, I think it might be more accurate to rephrase it to say "X series selected from K collections/L cases/M studies". The way it is phrased right now seems to suggest that entire collections/cases/studies were selected. Also, I would consider showing cart summary not in the tooltip, but next to the button. We do not have any use to that whitespace, and we do show content of a cohort when selected using the filter in the panel - not in the tooltip, so this would be more consistent.

<img width="382" alt="image" src="https://github.com/user-attachments/assets/4d924a14-a074-4365-ab6b-71bfa33f3451">
* Can we add size of the cart selection in TB? This would be useful, and consistent with the cohort selection.
* trying to download the manifest for the the cart shown above gave me "Internal Server Error - 500: There was an error while handling your request." Console did have more details:
```
 Error: Mismatched anonymous define() module: function($, utils) {


    // Resets forms in modals on hide. Suppressed warning when leaving page with dirty forms
    $('.modal').on('hide.bs.modal', function () {
        if(!$(this).prop("saving")) {
            if($(this).find('form').get().length) {
                $(this).find('form').get(0).reset();
            }
        }
    });

    $.getCookie = utils.getCookie;
    $.setCookie = utils.setCookie;
    $.removeCookie = utils.removeCookie;

    return {
        blacklist: /<script>|<\/script>|!\[\]|!!\[\]|\[\]\[\".*\"\]|<iframe>|<\/iframe>/ig,
        // From http://www.regular-expressions.info/email.html
        email: /^[a-z0-9!#$%&'*+/=?^_`{|}~-]+(?:\.[a-z0-9!#$%&'*+/=?^_`{|}~-]+)*@(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?$/,
        showJsMessage: utils.showJsMessage,
        // Simple method for standardizing storage of a message into sessionStorage so it can be retrieved and reloaded
        // at document load time
        setReloadMsg: function(type,text) {
            sessionStorage.setItem("reloadMsg",JSON.stringify({type: type, text: text}));
        },
        setCookie: function(name,val,expires_in,path) {
            utils.setCookie(name,val,expires_in,path);
        },
        removeCookie: function(name, path) {
            utils.removeCookie(name, path);
        },
        blockResubmit: utils.blockResubmit
    };
}
http://requirejs.org/docs/errors.html#mismatch
    at makeError (require.js:166:17)
    at intakeDefines (require.js:1236:36)
    at require.js:1423:25Understand this error
/explore/manifest/js/utils.js:1 
        
        
       Failed to load resource: the server responded with a status of 404 ()Understand this error
manifest/:1 Refused to execute script from 'https://expr-dot-idc-dev.appspot.com/explore/manifest/js/utils.js' because its MIME type ('text/html') is not executable, and strict MIME type checking is enabled.Understand this error
require.js:141 Uncaught Error: Script error for: utils
http://requirejs.org/docs/errors.html#scripterror
    at makeError (require.js:166:17)
    at HTMLScriptElement.onScriptError (require.js:1696:36)Understand this error
/explore/manifest/js/libs/underscore-min.js:1 
        
        
       Failed to load resource: the server responded with a status of 404 ()Understand this error
manifest/:1 Refused to execute script from 'https://expr-dot-idc-dev.appspot.com/explore/manifest/js/libs/underscore-min.js' because its MIME type ('text/html') is not executable, and strict MIME type checking is enabled.Understand this error
require.js:141 Uncaught Error: Script error for: underscore
http://requirejs.org/docs/errors.html#scripterror
    at makeError (require.js:166:17)
    at HTMLScriptElement.onScriptError (require.js:1696:36)Understand this error
/explore/manifest/js/libs/jquery-3.7.1.min.js:1 
        
        
       Failed to load resource: the server responded with a status of 404 ()Understand this error
manifest/:1 Refused to execute script from 'https://expr-dot-idc-dev.appspot.com/explore/manifest/js/libs/jquery-3.7.1.min.js' because its MIME type ('text/html') is not executable, and strict MIME type checking is enabled.Understand this error
require.js:141 Uncaught Error: Script error for: jquery
http://requirejs.org/docs/errors.html#scripterror
    at makeError (require.js:166:17)
    at HTMLScriptElement.onScriptError (require.js:1696:36)Understand this error
require.js:141 Uncaught Error: Load timeout for modules: jqueryui,bootstrap,session_security,assetscore,assetsresponsive,tablesorter
http://requirejs.org/docs/errors.html#timeout
    at makeError (require.js:166:17)
    at checkLoaded (require.js:692:23)
    at require.js:713:25Understand this error
manifest/:1         
        
       Failed to load resource: the server responded with a status of 500 ()
```
* nitpick: why is the cart page theme in purple, while the explore page is in black (header, icons color). Is this intentional? I would expect the color schemes to be synchronized.
<img width="450" alt="image" src="https://github.com/user-attachments/assets/700935ba-00b8-4ac9-9585-dd5fabd6c05b">
<img width="806" alt="image" src="https://github.com/user-attachments/assets/61825193-f0bd-4d92-81d1-ae4f60e5c8a6">
* after adding some number of collections to the cart, going to the cart to see the content gives me a lengthy spinner followed by the error.
<img width="471" alt="image" src="https://github.com/user-attachments/assets/0083f1cd-0b7c-4c20-a633-7b9de5fcac19">
<img width="440" alt="image" src="https://github.com/user-attachments/assets/d4c7c49a-4a5c-4293-86c0-91ca45f5d026">

**Comment by G-White-ISB** [comment]
1. We can rephrase the cart tooltip as you like. We could also move the cart summary out of a tooltip and onto the page. But I should mention we had previously considered putting the cohort summary in a tooltip. We can include file sizes.

2. Sorry I should have mentioned that cart manifest download was not working. It needs to be integrated with changes Suzanne is making to cohort manifest download

3. Please compare the cart page to the collections page and the cohorts page. They share the same style. explore page is different because it does not consist of a single table like these other pages

4. I am trying to track down that error you received fetching the cart data. If you can replicate please let me know.

**Comment by fedorov** [comment]
> We can rephrase the cart tooltip as you like. We could also move the cart summary out of a tooltip and onto the page. But I should mention we had previously considered putting the cohort summary in a tooltip. We can include file sizes.

Ok, let's rephrase the tooltip as I suggested and include total size: ""X series selected from K collections/L cases/M studies. Total size N GB/TB." 

> Please compare the cart page to the collections page and the cohorts page. They share the same style. explore page is different because it does not consist of a single table like these other pages

Let me rephrase the question. Is there any logic in having table in pink in the cart page, and black in the explore page? I have nothing against pink. Just curious about the motivation for using different colors.

> I am trying to track down that error you received fetching the cart data. If you can replicate please let me know.

I've just tried again, and was able to reproduce it on the first try by adding NLST collection to the cart.

**Comment by fedorov** [comment]
Few nitpicks:
1. replicate cart summary on the cart page
2. add button "Open cart" with the card icon in the upper right
3. fix tall line height in the cart page

**Comment by G-White-ISB** [comment]
@fedorov, all items in the comment directly above have been fixed. Also, the cart summary text has been changed to your recommendation, except size is not explicitly calculated. The size calculation is being considered.

**Comment by fedorov** [comment]
As communicated on slack, trying to add NLST to the cart results in a ~20 sec delay followed by empty cart.

Regarding the revised design, here's the initial mockup (colors/icons are placeholders, open for revision) (source https://drive.google.com/file/d/1UDZlab52Q1Ly3Q-j7kykmHgVqgs5oKsn/view?usp=sharing)

![image](https://github.com/user-attachments/assets/d02621d1-03a9-46a9-bfb2-e29e84d011c0)

**Comment by G-White-ISB** [comment]
What is that icon just to the LHS of the Filter cohort definition? Is that clickable?

**Comment by fedorov** [comment]
No, it's not a button, just an icon. I just had this idea to connect the concept of "filter cohort" with an intuitive icon, to parallel that for the cart cohort. Just an idea.

**Comment by fedorov** [comment]
Another observation/concern. If I try to remove an individual series from the cart after adding the entire NLST collection, after I click the remove from cart button at the series level, I do not get any spinner, but I also cannot interact with the page for about 25 seconds. At the very least, there should be a spinner, no?

**Comment by fedorov** [comment]
@G-White-ISB following up on the discussion this week, please return the capability that allows to expand a row in a table by clicking anywhere in the row (except the clipboard copy buttons etc) to make the new behavior consistent with the current behavior of the portal.

**Comment by fedorov** [comment]
@s-paquette bringing back the functionality to allow clicking on the row should not be hard. If it turns out to be difficult, we will revisit.

**Comment by fedorov** [comment]
Tooltips for cart should depend on the level. For collection level it is currently shows "add series".

<img width="292" alt="image" src="https://github.com/user-attachments/assets/c621aefa-2ee1-4ecf-b7aa-26a3ad666fbe">

**Comment by fedorov** [comment]
Spinner on operations is still missing (e.g., removing a single series from the cart).

Sometime spinner is present, then disappears, but some call is still not done.

**Comment by s-paquette** [comment]
@fedorov Yes these are largely all related at this point.

Builds are now up on dev and test which address the following:

- Larger chevron
- Chevron background is colored based on hover and open/closed status
- 'large' (i.e. via BQ) cart downloads available on cart page

Investigating: single-series addition seems to break under some conditions

Question: @fedorov In the case where eg. a study, case, or collection is added to the cart, but some of the series contained in those do not match the filter (eg. perhaps their modality is different) - does the filter take precedence over the cart addition?

**Comment by fedorov** [comment]
Good question! Let's add this to the agenda for the tomorrow meeting. I think it can be one way or another, and I am not sure which way is more intuitive.

**Comment by G-White-ISB** [comment]
With respect to this question: on June 20 in the SLACK releases-dev Channel:

George White: 
"But also.. if you filter first, say on Modality = Radiotherapy Structure Set, and then click the cart in the 4D_Lung row you will only add the 1600 studies in that collection associated with that modality. That is how TCIA works. Alternatively one could decide that clicking the cart for 4D_Lung adds ALL 6690 in that collection REGARDLESS of the current filter definition. I'm not sure which behavior would be more expected"

David Gibbs: "I think we should stick with our current system, making selections available as defined by the current filter.  1. Set filter, 2. make selections under that filter." 
Andrey Fedorov gives a thumbs up to David Gibbs comment.

**Comment by s-paquette** [comment]
@fedorov @pgundluru New build on dev and test which addresses some of the button moving/adding, etc. 

Andrey, to make sure I have this right, you want the cart info and filter info int their own distinct panels, right? We could swap the filter panel to behave like PDC's: they don't have a title on it, instead while there's no filter chosen it just says 'select a filter at the left' and once you do, the action buttons and filter def appear. Does this sound preferable? It saves us some space and looks a bit cleaner overall.

Then, for the cart, similarly while it's empty it's just a bit of text saying 'cart empty. use the cart icons in the table below to add series to this cart' or similar. Then once something is added, the descriptive text and buttons appear.

**Comment by fedorov** [comment]
> Andrey, to make sure I have this right, you want the cart info and filter info int their own distinct panels, right?

Yes, I would like to experiment with that, if this is not difficult. Can you do this, so I can tell if this looks good? Please ping me on slack if you can when it is ready.

> We could swap the filter panel to behave like PDC's: they don't have a title on it, instead while there's no filter chosen it just says 'select a filter at the left' and once you do, the action buttons and filter def appear. Does this sound preferable? It saves us some space and looks a bit cleaner overall.
> Then, for the cart, similarly while it's empty it's just a bit of text saying 'cart empty. use the cart icons in the table below to add series to this cart' or similar. Then once something is added, the descriptive text and buttons appear.

Yes, let's try this out!

Couple other cosmetic suggestions that I hope are quick to implement:

* Could you try adding a download button for both the cart and cohort filter along with the buttons/filter definition? Perhaps using the same icon we use down at the study level for downloading? Right now the download buttons are in the top right and separate from the cohort/cart definition, which is not intuitive. We can keep or remove the current cohort/cart download buttons.
* I suggest we replace the text "Filter definition" with "Cohort filter definition". Otherwise, "Filter Definition" text is disconnected with "Cohort Download" button.

**Comment by s-paquette** [comment]
@fedorov A note, the screencast has no audio for me; I don't know if it was supposed to, but I don't hear anything. It looks like in the video that you started with a cart already selected--can you let me know what series was in there before you started this? (I can't always reliably reproduce the behaviors without following the exact route taken, so it's best if I can start from the cart you have here in the video.)

The rest of this is all very doable. One point, I'd mentioned removing 'Filter Definition' entirely, since we're splitting cart contents and filter definition from one another there's maybe less reason to title anything. Did you want to keep those titles per your second bullet item?

**Comment by fedorov** [comment]
> @fedorov A note, the screencast has no audio for me; I don't know if it was supposed to, but I don't hear anything. It looks like in the video that you started with a cart already selected--can you let me know what series was in there before you started this? (I can't always reliably reproduce the behaviors without following the exact route taken, so it's best if I can start from the cart you have here in the video.)

There was no audio. Here's a new video that starts from empty: https://app.screencast.com/DgSCfrnFkCESH. In summary: 1) add single case from NLST to the cart; 2) observe enormous (~20 sec!) delays in updating the content of the collections list after resetting/changing the filter (and that is ~20 sec **after** the spinner disappears). 

> I'd mentioned removing 'Filter Definition' entirely, since we're splitting cart contents and filter definition from one another there's maybe less reason to title anything. Did you want to keep those titles per your second bullet item?

If we move the download button into the same tile as the cohort filter definition, then we can remove the title. Otherwise I think it is too difficult to connect the unnamed filter definition with the "Download cohort" button.

**Comment by s-paquette** [comment]
@fedorov Got it, and thanks for the new video. Here's what I have at the moment:

![image](https://github.com/user-attachments/assets/6abbe06e-1c47-48c8-a21e-38732d3db494)

One catch. View cart, unlike ALL the other buttons, redirects to a new page. Making it open a dialog would only slow the page down more, and making it open in a new tab (as opposed to the current one) would take some time to rewrite. Are we okay with this?

An upshot/stopgap is you can hit the back button from the cart and you will be back on the explorer page with no changes or reloads.

**Comment by fedorov** [comment]
> Here's what I have at the moment:

I feel like something is missing. Since we have 2 tiles, I feel like they need some titles. But I don't want to delay. We can refine later.

BUT few small things would be nice to fix:
* color of text with the content of cohort/cart should be consistent (black)
* cart empty button does not work
* can we change the text to "Filter cohort content: [blah]" and "Cart content: [blah]", and also make sure that "[blah]" is formatted consistently between cohort and cart? I suggest this format: "X series from Y collections / Z cases / N studies". Right now the order, capitalization and separators are different between the two.
* charts info button is still there in the collapsed charts panel

> View cart, unlike ALL the other buttons, redirects to a new page. [...] Are we okay with this?

Yes, in spirit of being agile, I am ok with this!

**Comment by s-paquette** [comment]
@fedorov Note, we don't currently tabulate the collections which are in a cohort. Although we can this requires some changes to how counts are done on the backend, so I'd like to recommend we break that specific item out into a separate ticket ('include number of collections in cohort stats' or similar) and otherwise I can make them all look the same.

**Comment by fedorov** [comment]
Ok, let's then align formatting while skipping collection count and total size for the cart content.

**Comment by s-paquette** [comment]
@fedorov Most recent build has addressed: info tooltip icon being hidden when the charts are collapsed, formatting and colors on the cart and cohort information text, button activities.

Remaining: removing/adding a specific series is causing an error when you attempt to view the cart, and the slowdowns with large carts.

**Comment by s-paquette** [comment]
@fedorov @pgundluru Dev tier now has the following:
* When the page first loads, the shift from Alphabetical sorting to Numeric will engage the spinner to prevent confusing clicks. Note that on a sufficiently fast browser or machine this can result in a little bit of a flicker.
* Expanding table contents will now engage the spinner
* Clicking on the cart icon to add or remove collections, cases, studies, and/or series should engage the spinner. I tested this using NLST, which was the slowest performing when adding/removing from the cart. Note the slowness is still present, and I have ideas on how to improve that. Let me know if you don't see a spinner when adding to/removing from the cart, as it wasn't consistent in my initial tests but seems to be now.
* Cart manifest downloads should be working, both as immediate downloads and delayed downloads for larger manifests.

**Comment by s-paquette** [comment]
Planned for when I get back: basic visual adjustments so it's not just the monolithic spinner no matter what anyone does, i.e., 'Updating cart...' for cart operations, 'Updating tables...' for table operations, this kind of thing.

**Comment by fedorov** [comment]
@s-paquette sounds great - let's advance to test!

**Comment by fedorov** [comment]
@s-paquette it appears that there is a long delay after resetting cohort filter and updating the content of the tables. There is no spinner while that is underway.

**Comment by s-paquette** [comment]
@fedorov I've been trying to replicate this on test and don't seem to be able to. How many patients/studies have you added before you reset the cart? If I add some and then reset the filter definition, the spinner comes up, and doesn't go down until the update is complete.

**Comment by fedorov** [comment]
I've just tried, and after clicking to add second patient from the patient table, I hit this error

<img width="2038" alt="image" src="https://github.com/user-attachments/assets/67fbe2dd-8f3d-4a4d-92c5-67bdf5428a7b" />

**Comment by fedorov** [comment]
Test manifest download in situations where manifest is just below 64K series to observe the time between disappearance of "Preparing ..." popup and file save dialog. This is not critical to hold release, but should be fixed in the future (if possible) - eg by adjusting the threshold to lower than 64K.

**Comment by fedorov** [comment]
As reported by @dclunie and confirmed by me:

---

https://meet.google.com/pce-adsa-ehs?authuser=0#:~:text=https%3A//expr%2Ddot%2Didc%2Ddev.appspot.com/

Selected:

NLST
NLST 100002

Tried to "download" one study 1.2.840.113654.2.55.187766322555605983451267194286230980878

Using s5cmd (not idc-index).

Pressed "Download s5cmd manifest"

Started to prepare manifest but after a long time got "There was an error generating your manifest. Please contact the administrator." at the op of the page.

**Comment by G-White-ISB** [comment]
Thanks for the bug report wrt study manifest. I am looking into it

**Comment by fedorov** [comment]
@G-White-ISB how difficult would it be to adjust the widths of the columns? Currently, colums that do not need to be wide that show counts are wide, and those where we need more space for the column header ("# of series in cart" and Series description) are narrow.

<img width="852" alt="Image" src="https://github.com/user-attachments/assets/0c5d8a4d-ecbf-44ae-a2e8-6754b9d13409" />

<img width="842" alt="Image" src="https://github.com/user-attachments/assets/e9d23cb0-3d01-464e-9f74-0e50aeb6d997" />

If this is not trivial, we should address this separately from this cart issue.

**Comment by fedorov** [comment]
@G-White-ISB +/- selectors that we have in prod are gone in expr

Prod:

<img width="451" alt="Image" src="https://github.com/user-attachments/assets/60d58267-326d-4435-ab58-51c864290aba" />

Expr:

<img width="420" alt="Image" src="https://github.com/user-attachments/assets/30cccc70-475a-4525-8cda-78ddc5ef3562" />

**Comment by fedorov** [comment]
Tooltip for the cart button is incorrect

<img width="414" alt="Image" src="https://github.com/user-attachments/assets/0982f512-d833-4801-acc5-ede11b7984e1" />

**Comment by Gibbsdavidl** [comment]
Regarding the +/- buttons... perhaps we don't want users putting all collections in the shopping cart?

Regarding the tooltip: I wasn't able to reproduce that bug.

**Comment by fedorov** [comment]
> Regarding the +/- buttons... perhaps we don't want users putting all collections in the shopping cart?

The functionality that those buttons do right now - before cart - is expand all of the items in the current table. I think it makes sense to keep the buttons with the same functionality.

> Regarding the tooltip: I wasn't able to reproduce that bug.

100% reproducible for me.

![Image](https://github.com/user-attachments/assets/e256c482-61ee-4d07-952b-72628e088574)

**Comment by pgundluru** [comment]
The arrow to close and expand in left hand side search scope is missing in Exp tier (Once you close the search scope area then there is no way to bring it back) 

https://github.com/user-attachments/assets/7d0f7978-6cb5-49a0-b021-bf4c8cd8c84c

**Comment by fedorov** [comment]
Status update:
* +/- button selectors fixed
* toltip fixed
* column widths adjusted

**Comment by fedorov** [comment]
@G-White-ISB I do not see the +/- buttons fixed in the test tier!

Plus the new issues identified today, replicated from slack here https://nciimagingdat-m8a6349.slack.com/archives/C045VQUAZ0S/p1741040448473179

![Image](https://github.com/user-attachments/assets/39884ca6-f0f2-44dc-a27c-91c4a3473dc8)

**Comment by pgundluru** [comment]
The

> The arrow to close and expand in left hand side search scope is missing in Exp tier (Once you close the search scope area then there is no way to bring it back)
> 
>  arrowtoexpand.mp4

This bug is resolved and tested in Exp Tier

**Comment by fedorov** [comment]
As of today morning, generation of a manifest with NLST as content fails completely.

![Image](https://github.com/user-attachments/assets/e6c9f81f-0d58-4930-beee-a2b8f5f9d80a)

**Comment by pgundluru** [comment]
Test tier - Large collections that are created as a non logged in user ( e.g. 77 TB size for Analysis results) are currently failing to download the IDC-Index manifests

**Comment by pgundluru** [comment]
Test tier after Suzanne pushed fixes 03-05 

Copy Button on URL dropdown is misplaced as well as when you open SQL button and select Copy

**Comment by s-paquette** [comment]
@pgundluru Can I get a picture of this? I'm not seeing it on my browser.

**Comment by pgundluru** [comment]
Selecting and expanding items in the cart is erroring with no other console clues. I selected PDMR collection to reproduce this 

![Image](https://github.com/user-attachments/assets/98ca2427-1b8f-431c-9a73-0a3ed64ced6e)

**Comment by Gibbsdavidl** [comment]
Windows 11, chrome and firefox.

1. In the filter, select TCGA collection
2. In the filter, select 'kidney' in primary site location
3. put all three collection level items into cart 
4. Cart contents: 6628 series from 3 collections / 941 cases / 1442 studies
5.  Hit download cart manifest, idc-index manifest, spinner spins
6. Returns, sits for a few seconds.  Then 502 Bad Gateway

**Comment by G-White-ISB** [comment]
> Selecting and expanding items in the cart is erroring with no other console clues. I selected PDMR collection to reproduce this
> 
> ![Image](https://github.com/user-attachments/assets/98ca2427-1b8f-431c-9a73-0a3ed64ced6e)

There was an issue with how SOLR was parsing 'None' values. Yep, that again. This is resolved in latest code check-in. Code is not yet deployed to dev or test, but is on expr.

**Comment by G-White-ISB** [comment]
> Windows 11, chrome and firefox.
> 
> 1. In the filter, select TCGA collection
> 2. In the filter, select 'kidney' in primary site location
> 3. put all three collection level items into cart
> 4. Cart contents: 6628 series from 3 collections / 941 cases / 1442 studies
> 5. Hit download cart manifest, idc-index manifest, spinner spins
> 6. Returns, sits for a few seconds.  Then 502 Bad Gateway

The cart manifest construction was timing out ~ 2 minutes. I've updated the cart manifest so that it is faster, comparable to filter manifest for a similar filter. Code is checked in but not deployed in dev or test.

**Comment by s-paquette** [comment]
@G-White-ISB New issue identified: If I select a single series from a given collection/study, then try to open a cases table for a different collection, the system errors out.

To reproduce:

1. Open the explorer page, drill down on a collection of your choice to the series level and add one series to the cart.
2. Attempt to expand another collection's cases. The bug should trigger.

**Comment by fedorov** [comment]
This is live, for any new problems identified let's open new issues.
